### PR TITLE
lsp: Auto update examples index

### PR DESCRIPTION
**Pull request by charlieegan3**
As pages are added to docs.styra.com/opa/rego-by-example, this json file will be updated meaning more keywords and builtins will be linked in the Regal hover messages.

### CODE REVIEW

File: internal/lsp/examples/index.json [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-        "time.clock": "builtins/time/clock",
-        "time.format": "builtins/time/format",
-        "time.now_ns": "builtins/time/now_ns",
-        "time.parse_ns": "builtins/time/parse_ns"
-    },
-    "keywords": {
-        "contains": "keywords/contains",
-        "some": "keywords/some"
-    }
+  "builtins": {
```
Comment (charlieegan3) [inline internal/lsp/examples/index.json:2]: Changes here will be updated when the workflow is manually run on main, or on schedule.

File: build/workflows/update-example-index/process.rego [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	some p in _pages
+	p[0] == "builtins"
+
+	l := count(p)
+
+	l == 2
+
+	name := p[1]
+
+	not count({p |
```
Comment (anderseknert) [inline build/workflows/update-example-index/process.rego:26]: `not count` looks fishy... even if it's the `>` being negated. `< 2` instead?

**Review by anderseknert — COMMENTED** [review]
Just a few minor remarks, but looks good!

File: .github/workflows/update-example-index.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write # this workflow updates repo contents
+    steps:
+      - name: Check out code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Install xq (part of yq)
+        run: |
+          pip install yq
```
[the remark this answers is not in the document]
Comment (anderseknert) [reply .github/workflows/update-example-index.yaml:23]:
```
➜ cat p.xml | rq -i xml --indent "  "
{
  "root": {
    "p": {
      "a": "1",
      "b": "2",
      "c": "3"
    }
  }
}
```

File: build/workflows/update-example-index/process.rego (hunk 2 above)
Comment (charlieegan3) [reply build/workflows/update-example-index/process.rego]: Hmm, that doesn't look good... I have fixed to your version now.

File: .github/workflows/update-example-index.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (charlieegan3) [reply .github/workflows/update-example-index.yaml]: Interesting! Have fixed this now to use rq and the cached binary too.

**Comment by charlieegan3** [comment]
Think all addressed here now so going to merge for a final test on main.

**Pull request merged**